QgisModelBaker/libqgsprojectgen/dataobjects/legend.py

```diff
@@ -73,9 +75,10 @@ def create(self, qgis_project: QgsProject, group=None):
                 item.create(qgis_project, subgroup)
             else:
                 layer = item.layer
-                if layer.name() not in existing_layer_names:
-                    index = get_suggested_index_for_layer(layer, group, self.ignore_node_names) if layer.isSpatial() else 0
+                if layer.dataProvider().dataSourceUri() not in existing_layer_source_uris:
+                    index = static_index if self.static_sorting else get_suggested_index_for_layer(layer, group, self.ignore_node_names) if layer.isSpatial() else 0
```

Comment:
Hm... I'm a little confused but still not sure if you are right. In my opinion what you suggest should look like this (read the ternary conditions from left to right):
```
  if self.static_sorting:
      index = static_index
  elif layer.isSpatial():
      index = get_suggested_index_for_layer(layer, group, self.ignore_node_names)
  else:
      index = 0
```
To have it from right to left, it  would be the case if there are brackets.
```
index = (static_index if self.static_sorting else get_suggested_index_for_layer(layer, group, self.ignore_node_names)) if layer.isSpatial() else 0
```
But even then, it would be like this (negating the first condition):
```
  if not layer.isSpatial():
      index = 0
  elif self.static_sorting:
      index = static_index
  else:
      index = get_suggested_index_for_layer(layer, group, self.ignore_node_names)
```

If you can confirm that, then I'm happy to replace the tenary condition. Because you showed me, that we spend both now to many time in struggling with it's readability :-)

Reply:
Heh, yes, I did misread it 🙈